### PR TITLE
Updated to sonar plugin api version 5

### PR DESCRIPTION
**Pull request by pingunaut**
sonar-plugin-api was updated to version 5. additionally all depricated methods and classes have been removed or replaced

### CODE REVIEW

**Comment by aheritier** [comment]
Hi @pingunaut 

  Thx a lot for this contribution. I have 2 remarks : 
  * About the style : Could you update your contribution without changing the code format of files. Several files diffs are unreadable (at least not easily). AFAIR the codebase was reformatted to follow SonarQube conventions : https://github.com/SonarSource/sonar-developer-toolset
  * About the goal : This update will make the plugin compatible only with SonarQube 5.0+ (AFAIU ?) while the current LTS is 4.5.2. I don't know if it's not too early. If few Sonar core devs (ping @bellingard @fmallet  @simonbrandhof) could give us some feedback about such change compared to the product roadmap.

Cheers

**Comment by pingunaut** [comment]
@aheritier Thanks al lot for your hints and resources! 
I updated the code with correct format now and downgraded plugin api version to 4.5.2 LTS.
I think you are right, pointing out that 5+ would be too early.

Best regards

**Comment by aheritier** [comment]
thanks a lot @pingunaut it is readable now. @bellingard @fmallet any recommendation/best practice about the minimal Sonar version to support in plugins ?

File: pom.xml [hunk 1]
```diff
@@ -1,10 +1,8 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <parent>
```
Comment (aheritier) [inline pom.xml]: @pingunaut Why did you remove the inheritance  ?

**Comment by bellingard** [comment]
Yep @aheritier => we try to keep compatibility with latest LTS (so 4.5 currently) as much as possible. When it's not possible, then we just use the version that we need.

File: pom.xml (hunk 1 above)
Comment (pingunaut) [inline pom.xml]: @aheritier I adopted the pom structure of sonar-reference-plugin, where parent is missing:
https://github.com/SonarSource/sonar-examples/blob/master/plugins/sonar-reference-plugin/pom.xml

But after checking it's content, I would add it again to keep the additional information. Are you OK with that?

Best Regards

Comment (aheritier) [inline pom.xml]: I have a doubt now :-)
I didn't know the reference template.
@bellingard what is your point of view ? what is the best practice ?

**Comment by pingunaut** [comment]
any news on that one? LTS version 4.5.2 was added. we are using this one since start of year in production with no issues detected so far, so seems to be stable.

**Comment by deepy** [comment]
Closing this since we're now targeting 6.7
